runtime/compiler/env/VMJ9.cpp

```diff
@@ -8642,8 +8720,40 @@ TR_J9SharedCacheVM::getResolvedMethods(TR_Memory *trMemory, TR_OpaqueClassBlock
    {
    TR::Compilation* comp = _compInfo->getCompInfoForCompOnAppThread() ? _compInfo->getCompInfoForCompOnAppThread()->getCompilation() : _compInfoPT->getCompilation();
    TR_ASSERT(comp, "Should be called only within a compilation");
-   if (((TR_ResolvedRelocatableJ9Method *) comp->getCurrentMethod())->validateArbitraryClass(comp, (J9Class *) classPointer))
+
+   bool validated = false;
+
+   if (comp->getOption(TR_UseSymbolValidationManager))
+      {
+      if (comp->getSymbolValidationManager()->inHeuristicRegion() ||
+          comp->getSymbolValidationManager()->getIDFromSymbol(static_cast<void *>(classPointer)))
+         validated = true;
+      else
+         TR_ASSERT_FATAL(false, "Class 0x%p should already be validated\n", classPointer);
+      }
+   else
+      {
+      validated = ((TR_ResolvedRelocatableJ9Method *) comp->getCurrentMethod())->validateArbitraryClass(comp, (J9Class *) classPointer);
```

Comment:
given that the J9Class * is a parameter, I'm not sure why we need to validate it here... No code in the compiler should do anything with a J9Class * in an AOT compilation unless it's been validated already (and, in this spot, you don't really know that it should be an arbitrary class validation anyway because you don't know where it came from).

Reply:
though i admit, the original code is broken in the same way...

Reply:
`validated = ((TR_ResolvedRelocatableJ9Method *) comp->getCurrentMethod())->validateArbitraryClass(comp, (J9Class *) classPointer);` is what the code was doing to begin with, so again, in the interest of minimizing the number of changes, I left it as is. You're right though, it shouldn't require validation (hence why the new code doesn't add a new record but rather checks that that symbol was already validated).